addons/product/models/product_pricelist.py

```diff
@@ -168,7 +171,8 @@ def _compute_price_rule(self, products, qty, uom=None, date=False):
             date = fields.Datetime.now()
 
         # Fetch all rules potentially matching specified products/templates/categories and date
-        rules = self._get_applicable_rules(products, date)
+        pricelistRule = self.env['product.pricelist.item']
+        rules = self._get_applicable_rules(products, date) if self else pricelistRule
```

Comment:
```suggestion
        ProductPricelistItem = self.env['product.pricelist.item']
        rules = self._get_applicable_rules(products, date) if self else ProductPricelistItem
```
according to guidelines (but with this version, I really think a `if not self: return self.env['product.pricelist.item']` in `_get_applicable_rules` would be even cleaner)